test/operations_test/operations_test.go

```diff
@@ -54,48 +54,71 @@ var _ = Describe("Operations", func() {
-				e.Set("operations.action_timeout", 5*time.Nanosecond)
-				e.Set("operations.mark_orphans_interval", 1*time.Hour)
+		postBrokerBody := func() Object {
+			brokerServer := NewBrokerServer()
+			ctx.Servers[BrokerServerPrefix+"123"] = brokerServer
+			return Object{
+				"id":         "123",
+				"name":       "test-broker",
+				"broker_url": brokerServer.URL(),
+				"credentials": Object{
+					"basic": Object{
+						"username": brokerServer.Username,
+						"password": brokerServer.Password,
+					},
+				},
 			}
+		}
 
-			ctx = NewTestContextBuilder().WithEnvPostExtensions(postHook).Build()
-		})
+		for isAsync := range []string{"true", "false"} {
+			When("job timeout runs out", func() {
+				BeforeEach(func() {
+					postHook := func(e env.Environment, servers map[string]FakeServer) {
+						e.Set("operations.action_timeout", 5*time.Nanosecond)
+					}
 
-		When("job timeout runs out", func() {
-			It("marks operation as failed", func() {
-				brokerServer := NewBrokerServer()
-				ctx.Servers[BrokerServerPrefix+"123"] = brokerServer
-				postBrokerRequestWithNoLabels := Object{
-					"id":         "123",
-					"name":       "test-broker",
-					"broker_url": brokerServer.URL(),
-					"credentials": Object{
-						"basic": Object{
-							"username": brokerServer.Username,
-							"password": brokerServer.Password,
-						},
-					},
-				}
+					ctx = NewTestContextBuilder().WithEnvPostExtensions(postHook).Build()
+				})
 
-				resp := ctx.SMWithOAuth.POST(web.ServiceBrokersURL).WithJSON(postBrokerRequestWithNoLabels).
-					WithQuery("async", "true").
-					Expect().
-					Status(http.StatusAccepted)
-
-				VerifyOperationExists(ctx, resp.Header("Location").Raw(), OperationExpectations{
-					Category:          types.CREATE,
-					State:             types.FAILED,
-					ResourceType:      types.ServiceBrokerType,
-					Reschedulable:     false,
-					DeletionScheduled: false,
-					Error:             "could not reach service broker",
+				It("marks operation as failed", func() {
+					resp := ctx.SMWithOAuth.POST(web.ServiceBrokersURL).WithJSON(postBrokerBody()).
+						WithQuery("async", "true").
+						Expect()
+
+					VerifyOperationExists(ctx, resp.Header("Location").Raw(), OperationExpectations{
+						Category:          types.CREATE,
+						State:             types.FAILED,
+						ResourceType:      types.ServiceBrokerType,
+						Reschedulable:     false,
+						DeletionScheduled: false,
+						Error:             "could not reach service broker",
+					})
 				})
 			})
-		})
+
+			When("reconciliation timeout runs out", func() {
+				BeforeEach(func() {
+					postHook := func(e env.Environment, servers map[string]FakeServer) {
+						e.Set("operations.reconciliation_operation_timeout", 2*time.Nanosecond)
+					}
+
+					ctx = NewTestContextBuilder().WithEnvPostExtensions(postHook).SkipBasicAuthClientSetup(true).Build()
+				})
+
+				It("marks operation as failed", func() {
```

Comment:
Like I said above, I don't see this as a realistic case for a REST call to fail with this, but I guess it's fine to test it like so

Reply:
i added a commit where the error is added to the operation errors if the operation exists. since operations are returned to end users, operation errors have to be non-internal a.k.a. operation errors are http errors